lib/suite-builder.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+var _ = require('lodash'),
+    State = require('./state');
+
+function nothing() {}
+
+function makeSkipBrowser(browser) {
+    if (typeof browser === 'string') {
+        return {browserName: browser};
+    }
+
+    if (typeof browser !== 'object' || browser === null) {
+        throw new TypeError('suite.skip browsers must be strings or object');
+    }
+
+    if (!browser.browserName) {
+        throw new Error('browserName is not specified');
+    }
+
+    if (typeof browser.browserName !== 'string') {
+        throw new TypeError('browserName must be a string');
+    }
+
+    if ('version' in browser && typeof browser.version !== 'string') {
+        throw new TypeError('version must be a string');
+    }
+
+    return browser;
+}
+
+function notString(arg) {
+    return typeof arg !== 'string';
+}
+
+function argumentsToArray(args) {
+    if (args.length === 1 && Array.isArray(args[0])) {
+        return args[0];
+    } else {
+        return Array.prototype.slice.call(args);
+    }
+}
+
+function SuiteBuilder(suite) {
+    this.suite = suite;
```

Comment:
In previous implementation suite builder created functions inside constructor because suite needs to be private. This needed because if suite will be modified gemini may fail.